src/error.ts

```diff
@@ -1,34 +1,37 @@
+import type { JobID } from './types';
 import { getWorkerID } from './worker-id';
 
 /** Penumbra error class */
 export class PenumbraError extends Error {
   /** The download URL or job ID throwing an error */
-  public id: string | number;
+  public id: JobID;
 
   /** The worker ID associated with this error */
   public worker: number | null;
 
   /** Error message */
-  public message: string;
+  public override message: string;
 
   /**
    * Extend new Error
    * @param error - Error
    * @param id - ID
    */
-  constructor(error: string | Error, id: string | number) {
+  constructor(error: string | Error, id: JobID) {
     if (typeof error === 'string') {
       super(error);
       this.message = error;
     } else {
       const { message } = error;
       super(message);
-      Object.keys(error).forEach((key) => {
+      for (const key of Object.keys(error)) {
         if (key !== 'message') {
-          // eslint-disable-next-line @typescript-eslint/no-explicit-any
-          (this as any)[key] = (error as any)[key];
+          const descriptor = Object.getOwnPropertyDescriptor(error, key);
+          if (descriptor) {
+            Object.defineProperty(this, key, descriptor);
+          }
```

Comment:
I think I follow what's happening here, but I'm curious about this approach vs something like `Object.assign()`. Want to add a comment explaining why properties are being copied in this manner?

Reply:
There are a lot of other ways to patch the prototype in JS which are also pretty safe, but with extremely strict typescript in this repo this is only acceptable format. I can add a comment to that effect!